domain/application/state/application.go

```diff
@@ -1076,6 +1078,72 @@ func (st *ApplicationState) deleteSimpleUnitReferences(ctx context.Context, tx *
 	return nil
 }
 
+// GetSecretsForApplication returns the secrets owned by the specified application.
+func (st *ApplicationState) GetSecretsForApplication(
+	ctx domain.AtomicContext, appName string,
+) ([]*coresecrets.URI, error) {
+	app := applicationName{Name: appName}
+	queryStmt, err := st.Prepare(`
+SELECT sm.secret_id AS &secretID.id
+FROM secret_metadata sm
+JOIN secret_application_owner sao ON sao.secret_id = sm.secret_id
+JOIN application a ON a.uuid = sao.application_uuid
+WHERE a.name = $applicationName.name
+`, app, secretID{})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var (
+		dbSecrets secretIDs
+		uris      []*coresecrets.URI
+	)
+	if err := domain.Run(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err = tx.Query(ctx, queryStmt, app).GetAll(&dbSecrets)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Trace(err)
+		}
+		uris, err = dbSecrets.toSecretURIs()
+		return errors.Trace(err)
```

Comment:
Why do we do this here? Should this not be done outside, considering `GetAll(...)` fills a variable already captured by the closure's scope?